packages/inter-protocol/src/psm/psm.js

```diff
@@ -243,47 +247,75 @@ export const start = async (zcf, privateArgs, baggage) => {
     E(anchorBrand).getAmountShape(),
     E(stableBrand).getAmountShape(),
   ]);
-  const makeWantStableInvitation = () =>
-    zcf.makeInvitation(
-      wantStableHook,
-      'wantStable',
-      undefined,
-      M.split({ give: { In: anchorAmountShape } }),
-    );
-  const makeGiveStableInvitation = () =>
-    zcf.makeInvitation(
-      giveStableHook,
-      'giveStable',
-      undefined,
-      M.split({ give: { In: stableAmountShape } }),
-    );
 
-  const publicFacet = Far('Parity Stability Module', {
-    getMetrics: () => metricsSubscriber,
-    getPoolBalance: () => anchorPool.getAmountAllocated('Anchor', anchorBrand),
-    makeWantStableInvitation,
-    makeGiveStableInvitation,
+  const PSMI = M.interface('PSM', {
+    getMetrics: M.call().returns(M.remotable('MetricsSubscriber')),
+    getPoolBalance: M.call().returns(anchorAmountShape),
+    makeWantStableInvitation: M.call().returns(M.promise()),
+    makeGiveStableInvitation: M.call().returns(M.promise()),
+    ...publicMixinAPI,
   });
 
-  const getRewardAllocation = () => feePool.getCurrentAllocation();
+  const publicFacet = vivifyFarInstance(
+    baggage,
+    'Parity Stability Module',
+    PSMI,
+    {
+      getMetrics() {
+        return metricsSubscriber;
+      },
+      getPoolBalance() {
+        return anchorPool.getAmountAllocated('Anchor', anchorBrand);
+      },
+      makeWantStableInvitation() {
+        return zcf.makeInvitation(
+          wantStableHook,
+          'wantStable',
+          undefined,
+          M.split({
+            give: { In: anchorAmountShape },
+            want: M.or({ Out: stableAmountShape }, {}),
+          }),
+        );
+      },
+      makeGiveStableInvitation() {
+        return zcf.makeInvitation(
+          giveStableHook,
+          'giveStable',
+          undefined,
+          M.split({
+            give: { In: stableAmountShape },
+            want: M.or({ Out: anchorAmountShape }, {}),
+          }),
+        );
+      },
+      ...publicMixin,
+    },
+  );
+
+  // TODO why does this operation return an object with a single operation?
   const { makeCollectFeesInvitation } = makeMakeCollectFeesInvitation(
     zcf,
     feePool,
     stableBrand,
     'Stable',
   );
-  const creatorFacet = Far('Parity Stability Module', {
-    getRewardAllocation,
-    makeCollectFeesInvitation,
+
+  // These are internal so they do not yet have defensive programming
```

Comment:
I think this comment overstates the implied degree of unsafety. On integrity, the object before is defined using only two bespoke methods, both of which have zero arguments. One returns an allocation, which is a by-copy structure. The other returns an invitation, which we presume to already be defensive.

What the comment is about is that it isn't defined using interface guards, and so not defensive against extra arguments being provided that might take up space. Aside from space, these extra arguments are ignored, so cannot do any other damage. There's also nothing retaining the extra arguments, so they'll be gc-able immediately anyway, which is as fast as discarding them (currently) via interface guards.

Reply:
I agree. I'll update the comment if I have to redo anything but otherwise isn't worth a trip through CI.